controller/konnect/ops/server_url.go

```diff
@@ -1,13 +1,27 @@
 package ops
 
-import "strings"
+import (
+	"strings"
+
+	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
+)
 
 // ServerURL is a type to represent a server URL.
 type ServerURL string
 
 // NewServerURL creates a new ServerURL from a string. It accepts either a full URL or a hostname.
-func NewServerURL(u string) ServerURL {
-	// If the URL does not have an HTTPs scheme, we prepend it HTTPs.
+func NewServerURL[T any](u string) ServerURL {
+	// NOTE: Code below tries to determine if we need a global API endpoint for the given type.
+	// Thus far there's no way to determine that programmatically (easily).
+	var t T
+	switch any(t).(type) {
+	case konnectv1alpha1.KonnectCloudGatewayNetwork:
+		// TODO: add more types here that target the global API endpoint.
```

Comment:
Out of curiosity: do we already know of others?

Reply:
Not at this moment. There are some like ProviderAccounts https://docs.konghq.com/konnect/api/cloud-gateways/latest/#/Provider%20Accounts/list-provider-accounts but those we are not going to managed through k8s CRs as adding them is a manual process.